src/domain/products/edit.tsx

```diff
@@ -21,8 +25,10 @@ const EditProductPage = ({ id }) => {
   const notification = useNotification()
   const { product, isLoading } = useAdminProduct(id)
   const updateProduct = useAdminUpdateProduct(id)
+  const [submitting, setSubmitting] = useState(false)
 
   const onSubmit = async (data) => {
+    setSubmitting(true)
```

Comment:
this flag also needs to be set to `false` if an image upload fails for some reason (L:40) otherwise the form will be "stuck" in the submitting state.